packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/model/configurations/ModelConfiguration.java

```diff
@@ -0,0 +1,49 @@
+package software.uncharted.terarium.hmiserver.models.dataservice.model.configurations;
+
+import com.fasterxml.jackson.annotation.JsonManagedReference;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Transient;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+import software.uncharted.terarium.hmiserver.annotations.TSIgnore;
+import software.uncharted.terarium.hmiserver.annotations.TSModel;
+import software.uncharted.terarium.hmiserver.models.TerariumAsset;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@TSModel
+@Accessors
+@Entity
+public class ModelConfiguration extends TerariumAsset {
+	private UUID calibrationRunId;
+	private UUID modelId;
+
+	@Transient
+	private Map<String, Semantic> values;
+
+	@TSIgnore
+	@OneToMany(mappedBy = "modelConfiguration", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
```

Comment:
I think we may need to have a think about indices, the way the PG is setup, where you have a model-configuration table and a semantics table via a foreign key.

The number of entries in the Semantics table will greatly out-number the number of model-configurations ... say by a factor of 15 to 20. And the indices don't look like they are setup properly:

```
Indexes:
    "semantic_pkey" PRIMARY KEY, btree (id)
Check constraints:
    "semantic_type_check" CHECK (type >= 0 AND type <= 2)
Foreign-key constraints:
    "fkjx7n8qnbr2wnt3ar4j7gp8tkx" FOREIGN KEY (model_configuration_id) REFERENCES model_configuration(id)
Access method: heap
```

Reply:
We may have sufficient data, after migration, that we will end up with semantic table(s) with rows in the tens of thousands. So it can be bottleneck as is, since we bring out "suggested configurations" the scan will happen numerous times when we open up the model-config drilldown.

Given how often data migrations had annoyed us  in the past - I think it would be good to at least get some metrics and amend stuff where we can, to minimize things shifting around down the road.

Reply:
I did some perf tests, with around 50k entries devoted to configurations (initials + params + observables) we are still okay-ish without indices.

With my tests, joining model_config to semantics
- without index is about 15-20 milliseconds
- with index is about 2-3 milliseconds

Will leave it with @dvince2 if this is something we should pursue now.